### ISSUE TITLE
Allow unbuffered compileLog

### ISSUE THREAD

**Issue by silversquirl** [ziglang/zig#14711]
`@compileLog` is a very useful function for debugging comptime code. However, in some cases, comptime code can crash the compiler. In these instances, any compile logs will not be printed, due to the internal buffering Zig does.

For debugging compiler bugs, it would be extremely useful to have a command line flag such as `-fno-buffer-compile-log` that disables the buffering and instead outputs compile logs immediately, as soon as the `@compileLog` statement is hit.

**Comment by r00ster91** [comment]
Or can we make it so that `@compileLog`s are always printed even if comptime code crashed the compiler?

**Comment by matu3ba** [comment]
> Or can we make it so that @compileLogs are always printed even if comptime code crashed the compiler? Whenever there is any data buffered.

Sounds like a custom segfault handler for stage3+4 would do the job. Those are currently not signal safe (yet, see #14658), but they should be good enough.

**Comment by silversquirl** [comment]
Yep, that would also work but I'm not sure how annoying it'd be to implement. It would require storing the compileLog buffer in a global somewhere, which might not be ideal.

**Comment by mlugg** [comment]
> Yep, that would also work but I'm not sure how annoying it'd be to implement.

Easy actually, I wrote about 2 lines in the crash handler and they might work, will test shortly

**Comment by andrewrk** [comment]
It sounds like the use case here is debugging, in which case compile logs could simply have an additional log.debug() call and then one could use the --verbose-log flag to access them.
